packages/frontend/core/src/modules/app-sidebar/views/sidebar-header/sidebar-switch.tsx

```diff
@@ -16,15 +16,23 @@ export const SidebarSwitch = ({
 }) => {
   const appSidebarService = useService(AppSidebarService).sidebar;
   const open = useLiveData(appSidebarService.open$);
+  const preventHovering = useLiveData(appSidebarService.preventHovering$);
   const switchRef = useRef<HTMLDivElement>(null);
 
   const handleMouseEnter = useCallback(() => {
+    if (preventHovering) {
```

Comment:
You can start listening for the mouseenter event after the sidebar is collapsed, which should achieve the same effect without adding additional states.